cmd/query/app/query_parser.go

```diff
@@ -46,22 +49,68 @@ const (
 var (
 	errMaxDurationGreaterThanMin = fmt.Errorf("'%s' should be greater than '%s'", maxDurationParam, minDurationParam)
 
-	// ErrServiceParameterRequired occurs when no service name is defined
-	ErrServiceParameterRequired = fmt.Errorf("parameter '%s' is required", serviceParam)
+	// errServiceParameterRequired occurs when no service name is defined.
+	errServiceParameterRequired = fmt.Errorf("parameter '%s' is required", serviceParam)
+
+	jaegerToOtelSpanKind = map[string]string{
+		"unspecified": metrics.SpanKind_SPAN_KIND_UNSPECIFIED.String(),
+		"internal":    metrics.SpanKind_SPAN_KIND_INTERNAL.String(),
+		"server":      metrics.SpanKind_SPAN_KIND_SERVER.String(),
+		"client":      metrics.SpanKind_SPAN_KIND_CLIENT.String(),
+		"producer":    metrics.SpanKind_SPAN_KIND_PRODUCER.String(),
+		"consumer":    metrics.SpanKind_SPAN_KIND_CONSUMER.String(),
+	}
+)
+
+type (
+	// queryParser handles the parsing of query parameters for traces.
+	queryParser struct {
+		traceQueryLookbackDuration time.Duration
+		timeNow                    func() time.Time
+	}
+
+	traceQueryParameters struct {
+		spanstore.TraceQueryParameters
+		traceIDs []model.TraceID
+	}
+
+	dependenciesQueryParameters struct {
+		endTs    time.Time
+		lookback time.Duration
+	}
+
+	durationParser = func(s string) (time.Duration, error)
```

Comment:
converted the duration parsers to functions. I did initially try to adopt functional programming but was thrown by the fact that the unit duration parser needed to maintain state (the duration units), but hadn't thought that we can get around this via closures. 

Thanks for pointing this out, @yurishkuro, it looks a lot neater, and I've learnt something new! :)